### ISSUE TITLE
Some unsupported syntaxes

### ISSUE THREAD

**Issue by benjie** [oguimbal/pgsql-ast-parser#24]
Stellar job on pg-mem; it's already incredibly impressive! Your README requests issues for unsupported (unparseable) syntaxes; whilst trying to run [the PostGraphile introspection query](https://github.com/graphile/graphile-engine/blob/9940895db5a35f218484620d610845201278611c/packages/graphile-build-pg/src/plugins/introspectionQuery.js#L26-L431) (with placeholders replaced) against pg-mem I found the following issues:

- aliases starting with an underscore do not seem to be supported: `select 1 _one` / `select 1 as _one`
- empty arrays do not seem to be supported: `select array[]::text[];` (typically used as the fallback in a `COALESCE`)
- the `operator(...)` syntax is not supported (`select 1 + 2` is equivalent to `select 1 operator(pg_catalog.+) 2`; the latter is safer in that it prevents the `+` operator from being overridden by other operators in the search_path, such as those from the `intarray` extension)
  - I'd be happy if you just rewrote `operator(pg_catalog.SOMETHING)` to just `SOMETHING` perhaps earlier in the pipeline, I never reference operators explicitly from other schemas.
- the `&` operator doesn't seem to be supported; `select 24 & 9;`
- aliases for function sources don't work without `as` keyword:
  - `select 1 from fn() alias;` errors
  - `select 1 from fn() as alias;` parses
- nested CTEs don't parse:
  ```sql
  with a as (
    with b as (
      select 1
    ) select * from b
  ) select * from a;
  ```
- array constructor syntax (`array(...)`) does not seem to be supported:
  ```sql
  select array(
    select 1
    from pg_class
  );
  ```
- `union all` syntax is unsupported, though `union` syntax works
  ```sql
  select 1
  union all
  select 2;
  ```
- `union` in subqueries is unsupported:
  ```sql
  select i from generate_series(1, 10) as i where i in (
    select 1
    union
    select 2
  );
  ```

I hope this helps; keep up the excellent work! :raised_hands: 

(If you don't intend to support these syntaxes I will not be at all offended if you close this issue.)

**Comment by BrandiATMuhkuh** [comment]
The follow query is not working. `Unexpected lparen token: "(".` near `(COUNT(*) FILTER (`

You can find the documenation for it at point `4.2.7. Aggregate Expressions` in [this pg-doc page](https://www.postgresql.org/docs/9.4/sql-expressions.html)



```
SELECT
  "userId",
 (COUNT(*) FILTER (WHERE "correct" = 'YES'))::numeric AS "correct",
 FROM "My_Table"
```

**Comment by oguimbal** [comment]
Thanks... adding that to my todo list.

(Everything which is valid pg syntax is in scope... and implementing the parser is the easy part)

**Comment by oguimbal** [comment]
Should all be fixed in `pgsql-ast-parser@6.3.0`  except @BrandiATMuhkuh issue, which is now #25
